syntax = "proto3";
service Movie {
    rpc GetMovieByID(MovieID) returns (MovieData) {}
    rpc GetMovieByTitle(MovieTitle) returns (MovieData) {}
    rpc GetListMovies(Empty) returns (stream MovieData) {}
    rpc AddMovie(MovieData) returns (MovieData) {}
    rpc DeleteMovie(MovieID) returns (MovieData) {}
    rpc EditMovie(EditData) returns (MovieData) {}
}
message MovieID {
    string id = 1;
}
message EditData {
    string movieID = 1;
    float rating = 2;
}
message MovieTitle {
    string title = 1;
}
message MovieData {
    string title = 1;
    float rating = 2;
    string director = 3;
    string id = 4;
}
message Empty {}